cpp/unreal_plugins/UrdfBot/Source/UrdfParser.h

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <Eigen/Dense>
+#include <map>
+#include <vector>
+
+#include <CoreMinimal.h>
+
+class FXmlNode;
+
+struct UrdfJointDesc;
+struct UrdfLinkDesc;
+struct UrdfMaterialDesc;
+
+enum UrdfGeometryType
+{
+    INVALID_GEOMETRY = 0,
+    BOX,
+    SPHERE,
+    CYLINDER,
+    MESH
+};
+
+enum UrdfJointType
+{
+    // Some of these (such as FIXED) are already macros
+    INVALID_TYPE = 0,
+    REVOLUTE_TYPE,
+    CONTINUOUS_TYPE,
+    PRISMATIC_TYPE,
+    FIXED_TYPE,
+    FLOATING_TYPE,
+    PLANAR_TYPE
+};
+
+struct UrdfGeometry
+{
+    UrdfGeometryType type_;
+    FVector size_; // requried by BOX
+    float radius_; // requried by SPHERE, CYLINDER
+    float length_; // requried by CYLINDER
+    std::string file_name_; // requried by MESH
+};
+
+struct UrdfLinkDesc
+{
+    std::string name_;
+
+    // non-derived data (inertial, visual, etc)...
```

Comment:
Don't need this comment: `// non-derived data (inertial, visual, etc)...`

(I know this came from my original sketch but I don't think it's needed in the actual implementation)

Reply:
fixed